go/base/tasks.py

```diff
@@ -0,0 +1,49 @@
+import json
+from celery.task import task
+
+from go.base.utils import vumi_api
+from go.config import get_conversation_definition
+from go.vumitools.metrics import get_conversation_metric_prefix
+
+
+@task(ignore_result=True)
+def send_recent_conversation_metrics():
+    api = vumi_api()
+    try:
+        conversation_details = get_and_reset_recent_conversations(api)
+
+        for conv_details in conversation_details:
+            details = json.loads(conv_details)
+            user_api = api.get_user_api(details["account_key"])
+            conv = user_api.get_conversation(details["conv_key"])
+            publish_conversation_metrics(api, user_api, conv)
+    finally:
+        api.close()
+
+
+def get_and_reset_recent_conversations(vumi_api):
+    redis = vumi_api.redis.sub_manager("conversation.metrics.middleware")
+
+    # makes use of redis atomic functions to ensure nothing is added to the set
+    # before it is deleted
+    try:
+        redis.rename("recent_coversations", "old_recent_conversations")
+    except:
+        return []
```

Comment:
Bleh. This is annoying. The best solution I can think of is to sticking the appropriate exception class on the redis_manager so we can do `except redis.RESPONSE_ERROR`, but that requires more changes to core Vumi. For now, I suggest one of:
* Check whether `recent_conversations` exists first (since we shouldn't be calling the task twice simultaneously and only the task should delete it).
* Something messy like:
```python
except Exception as err:
    if type(err).__name__ != "ResponseError":
        raise
    return []
```
I'm leaning towards the first option, because relying on class names is meh (even though its how the Vumi redis tests check things).